packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -80,26 +172,56 @@ export class DomainName extends Resource implements IDomainName {
   public readonly name: string;
   public readonly regionalDomainName: string;
   public readonly regionalHostedZoneId: string;
+  private readonly domainNameConfigurations = new Array<CfnDomainName.DomainNameConfigurationProperty>();
 
   constructor(scope: Construct, id: string, props: DomainNameProps) {
     super(scope, id);
 
+    // domain name null check
     if (props.domainName === '') {
       throw new Error('empty string for domainName not allowed');
     }
 
+    // domain name configuration null check
+    if (!props.domainNameConfigurations) {
+      throw new Error('empty domain name configurations are not allowed');
+    } else {
+      this.setDomainNameConfigurations(...props.domainNameConfigurations);
```

Comment:
Just curious. Any reason for setting here? It is also readonly and AFAIK you cannot modify props in this class directly. So why not just use props.domainNameConfiguration directly?

Reply:
I couldn't directly map this array of objects to another array of objects due to the structure of DomainNameConfiguration. So, needed to copy over every item in the list.